CefSharp/Internals/JavascriptObjectRepository.cs

```diff
@@ -259,15 +329,48 @@ private static JavascriptMethod CreateJavaScriptMethod(MethodInfo methodInfo, bo
             return jsMethod;
         }
 
-        private static JavascriptProperty CreateJavaScriptProperty(PropertyInfo propertyInfo, bool camelCaseJavascriptNames)
+        private object ConvertIfGenericList(object value, Type type, bool camelCaseJavascriptNames)
+        {
+            if (!type.IsGenericType) return value;
+
+            IList list = value as IList;
+            if (list != null)
```

Comment:
Simplify this, evaluate `null` and throw exception straight away, reduces nesting.